esmvaltool/recipes/recipe_hyint.yml

```diff
@@ -0,0 +1,85 @@
+# recipe_hyint.yml
+---
+
+documentation:
+  description: |
+    Recipe for the HyInt package by E. Arnone and J. von Hardenberg (ISAC-CNR)
+
+  authors:
+    - arno_en
+    - hard_jo
+
+  references:
+    - giorgi11jc
+    - giorgi14jgr
+
+  projects:
+    - c3s-magic
+
+datasets:
+  - {dataset: ACCESS1-0,      project: CMIP5, exp: [historical,rcp85], ensemble: r1i1p1, start_year: 2004, end_year: 2006}
+#  - {dataset: ACCESS1-3,      project: CMIP5, exp: rcp85, ensemble: r1i1p1, start_year: 2006, end_year: 2016}
+#  - {dataset: MPI-ESM-MR,      project: CMIP5, exp: rcp85, ensemble: r1i1p1, start_year: 2006, end_year: 2016}
+
+preprocessors:
+  preproc:
+    extract_levels: false
+
+diagnostics:
+  hyint:
+    description: HyInt hydroclimatic indices calculation and plotting
+    variables:
+       pr:
+        preprocessor: preproc
+        reference_dataset: "ACCESS1-0"
+        mip: day
+        field: T2Ds
+    scripts:
+      main:
+        script: hyint/hyint.R
+        # Define whether model data should be regridded 
+        # a) false to keep original resolution
+        # b) set desired regridding resolution in cdo format e.g., "r320x160"
+        # c) "REF" to use resolution of reference model
+        rgrid: false   
```

Comment:
This should preferably be done with the preprocessor, is there a good reason why cdo is used now?

Reply:
We first need to evaluate climate indices on the original grid, then regrid on a common grid if required (e.g. to produce difference of fields from two models). So the preproc regridding cannot be used.